surfer/viz.py

```diff
@@ -1600,6 +1600,14 @@ def add_foci(self, coords, coords_as_verts=False, map_surface=None,
             If None, it is assumed to belong to the hemipshere being
             shown. If two hemispheres are being shown, an error will
             be thrown.
+        allow_data : bool | False
+            If True, will plot foci using quiver3d instead of points3d to allow
+            mapping data to spheres.
+        data : numpy array
+            1D array the same size as the number of foci. Spheres will be
+            colorcoded acording to data values, whereas spheres' sizes will be
+            coded using the absolute values of data.
```

Comment:
API seems quite complicated. Can you achieve what you want using only one new parameter? and without adding a new method?

also you miss tests and some new example for documentation.

Reply:
I agree with @agramfort. Can `allow_data` as an external parameter not be replaced with an internal check for `data is not None`?

Reply:
I also am not sure that I like forcing the color *and* size of the spheres by default. There are lots of cases when you might want colors to reflect a scalar but a uniform size. Or (probably less commonly) vice versa. Also I don't see any way to control the size range, colormap, etc. through this API, which would be important.

Reply:
Thanks for pointing out the issues, i'll try to fix them. This is my first attempt at a pull request, so bear with me, especially with the tests.